packages/ra-ui-materialui/src/input/DateInput.tsx

```diff
@@ -202,38 +205,65 @@ export type DateInputProps = CommonInputProps &
     Omit<TextFieldProps, 'helperText' | 'label'>;
 
 /**
- * Convert Date object to String
+ * Convert Date object to String, ignoring the timezone.
  *
  * @param {Date} value value to convert
  * @returns {String} A standardized date (yyyy-MM-dd), to be passed to an <input type="date" />
  */
 const convertDateToString = (value: Date) => {
```

Comment:
We still need this change for cases where the initial value is a non-standard date string (e.g. 02/02/2001)